dc-cudami-server/dc-cudami-server-business/src/test/java/de/digitalcollections/cudami/server/business/impl/service/identifiable/IdentifiableServiceImplTest.java

```diff
@@ -499,4 +499,113 @@ public void allowMultiplePrimariesForDifferentTuples()
+  @Test
+  public void updateWithPrimaryUrlAliasesOnly()
+      throws CudamiServiceException, IdentifiableServiceException, ValidationException {
+    UUID targetUuid = UUID.randomUUID();
+
+    // in DB (should be unset)
+    LocalizedUrlAliases storedUrlAliases = new LocalizedUrlAliases();
+
+    UrlAlias firstStoredPrimaryUrlAlias = new UrlAlias();
+    firstStoredPrimaryUrlAlias.setUuid(UUID.randomUUID());
+    firstStoredPrimaryUrlAlias.setCreated(LocalDateTime.now());
+    firstStoredPrimaryUrlAlias.setLastPublished(LocalDateTime.now());
+    firstStoredPrimaryUrlAlias.setPrimary(true);
+    firstStoredPrimaryUrlAlias.setTargetUuid(targetUuid);
+    firstStoredPrimaryUrlAlias.setTargetLanguage(Locale.GERMAN);
+    firstStoredPrimaryUrlAlias.setSlug("slug1");
+
+    UrlAlias secondStoredPrimaryUrlAlias = new UrlAlias();
+    secondStoredPrimaryUrlAlias.setUuid(UUID.randomUUID());
+    secondStoredPrimaryUrlAlias.setCreated(LocalDateTime.now());
+    secondStoredPrimaryUrlAlias.setLastPublished(LocalDateTime.now());
+    secondStoredPrimaryUrlAlias.setPrimary(true);
+    secondStoredPrimaryUrlAlias.setTargetUuid(targetUuid);
+    secondStoredPrimaryUrlAlias.setTargetLanguage(Locale.ENGLISH);
+    secondStoredPrimaryUrlAlias.setSlug("slug2");
+
+    storedUrlAliases.add(firstStoredPrimaryUrlAlias, secondStoredPrimaryUrlAlias);
+
+    when(urlAliasService.findLocalizedUrlAliases(eq(targetUuid))).thenReturn(storedUrlAliases);
+
+    // new ones
+    LocalizedUrlAliases localizedUrlAliases = new LocalizedUrlAliases();
+
+    UrlAlias firstPrimaryUrlAlias = new UrlAlias();
+    firstPrimaryUrlAlias.setPrimary(true);
+    firstPrimaryUrlAlias.setTargetUuid(targetUuid);
+    firstPrimaryUrlAlias.setTargetLanguage(Locale.GERMAN);
+    firstPrimaryUrlAlias.setSlug("slug-de-new");
+
+    UrlAlias secondPrimaryUrlAlias = new UrlAlias();
+    secondPrimaryUrlAlias.setPrimary(true);
+    secondPrimaryUrlAlias.setTargetUuid(targetUuid);
+    secondPrimaryUrlAlias.setTargetLanguage(Locale.ENGLISH);
+    secondPrimaryUrlAlias.setSlug("slug-en-new");
+
+    localizedUrlAliases.add(firstPrimaryUrlAlias, secondPrimaryUrlAlias);
+
+    Identifiable identifiable = new Identifiable();
+    identifiable.setUuid(targetUuid);
+    LocalizedText label = new LocalizedText(Locale.GERMAN, "label");
+    label.setText(Locale.ENGLISH, "label");
+    identifiable.setLabel(label);
+    identifiable.setLocalizedUrlAliases(localizedUrlAliases);
+
+    when(repo.update(eq(identifiable))).thenReturn(identifiable);
+    service.update(identifiable);
+
+    verify(urlAliasService, times(2)).create(any(), eq(true));
+    verify(urlAliasService, times(2)).update(any());
+    verify(urlAliasService).create(eq(firstPrimaryUrlAlias), eq(true));
+    verify(urlAliasService).create(eq(secondPrimaryUrlAlias), eq(true));
+    // the stored UrlAliases must have been changed to primary == false
+    assertThat(firstStoredPrimaryUrlAlias.isPrimary()).isFalse();
```

Comment:
Müßte der vor dem Assert nicht erstmal aus der DB gelesen werden? 
Wenn ich nichts übersehen habe, testest Du hier doch den oben angelegten Mock, und bei dem sollte `isPrimary()` eigentlich `true` sein

Reply:
Hier gibt's keine DB. Diese Objekte kommen via Mock in `update` rein. Dort wird `primary` auf `false` gesetzt und die Objekte werden wieder in die DB geschrieben (via urlAliasService). Weil's aber ja ein und dasselbe Objekt bleibt, kann man hier *nach* dem Aufruf von `update` die Änderung sehen.  
*vor* `service.update(...)` -> primary == true  
*danach* -> primary == false

Reply:
OK, call by reference... Dann paßts.